src/Storages/Kafka/StorageKafka.cpp

```diff
@@ -966,9 +966,12 @@ void registerStorageKafka(StorageFactory & factory)
         {
             throw Exception(ErrorCodes::BAD_ARGUMENTS, "kafka_poll_max_batch_size can not be lower than 1");
         }
-        if (args.columns.getOrdinary() != args.columns.getAll() || !args.columns.getWithDefaultExpression().empty())
+        // Filter Default column type and skip Materialized, Ephemeral and Alias types.
```

Comment:
The comment in l. 969 just paraphrases the code in l. 970 (but in a confusing way). Maybe replace it \+ the comment on l. 971 by simply: `Kafka engine allows only ordinary columns without default expression or alias columns.`.

(and perhaps it would be more intuitive and robust to invert the check, i.e. instead of checking what type the columns must NOT be, check that they are all of the desired type)

Reply:
Simplified predicate and removed obsolete function `ColumnsDescription::getWithDefaultExpression` (It was used only for KafkaEngine).